doc/timing.adoc

```diff
@@ -0,0 +1,222 @@
+----
+minetest.register_globalstep(function(dtime)
+	-- do something, probably involving dtime
+end)
+----
+
+==== Params
+[%autowidth, frame=none]
+|===
+| `dtime` | number | Delta (passed) time since last step in seconds
+|===
+
+TIP: Use globalsteps to poll for an event for which there are no callbacks, such as player controls (`player:get_player_control()`).
+
+WARNING: As globalsteps run every server step, they are highly performance-critical and must be well-optimized. If globalsteps take longer than the server step to complete, the server thread is blocked and the server becomes "laggy".
+
+=== `minetest.after(time, func, ...)`
+
+Scheduling. Calls `func(...)` after at least `time` seconds have passed.
+
+TIP: Use `minetest.after(0, func)` to immediately do load-time stuff that is only possible at run-time, or to schedule something for the next server step.
+
+==== Usage
+[source,lua]
+----
+job = minetest.after(time, func, ...)
+if ... then
+	job:cancel()
+end
+----
+
+==== Params
+[%autowidth, frame=none]
+|===
+| `time` | number | Time in seconds that must have passed for the callback to be executed.
+| `func` | callable | Function to be called. Objects with the `__call` metamethod are supported as well.
+| `...` | vararg | Arguments to be passed to `func` when it is called.
+|===
+
+==== Returns
+[%autowidth, frame=none]
+|===
+| `job` | job object | Simple object providing a `job:cancel()` method to cancel a scheduled "job".
+|===
+
+WARNING: `...` may be arbitrarily cut off at `nil` values, as Minetest uses a simple list to store the args. Don't include `nil`s in the arguments if possible.
+
+TIP: If you have to call a function with `nil`s in it's argument list, use a closure for reliable behavior:
+[source,lua]
+----
+minetest.after(time, function()
+	func(nil, arg, arg2, nil, nil)
+end)
+----
+
+WARNING: All scheduled callbacks are stored in a list until they are called. This list is traversed in linear time if *any* of the callbacks is executed. For this reason, don't use `minetest.after` excessively.
+
+WARNING: `job:cancel()` might cause a memory leak if used naively: The job - including the arguments `...` - is still kept in the job list until the job expires; only the function call isn't made. Example: `minetest.after(math.huge, error, "a super long string"):cancel()` - the string will stay in memory forever despite the job being cancelled, as `math.huge` is never reached.
+
+== Entities
+
+=== `entity:on_step(dtime, ...)`
+
+Callback. Called per-entity every globalstep with the current `dtime`, allowing for comfortable per-entity timing.
+
+==== Usage
+[source,lua]
+----
+function entity:on_step(dtime, ...)
+	-- use dtime
+end
+----
+
+==== Params
+[%autowidth, frame=none]
+|===
+| `self` | entity | The entity itself (implicit parameter)
+| `dtime` | number | Delta (passed) time since last step in seconds (same as globalstep)
+| `...` | vararg | Other parameters irrelevant to timing (such as `moveresult`)
+|===
+
```

Comment:
There are obviously intersections. But for this to be a complete overview concerning timing, entity steps must be mentioned, but not with great detail.

Reply:
Fair, perhaps this can be turned into a linked/included section when the entity specific documentation becomes a thing. We'll cross that bridge when we come to it though.